pkg/aws/cloud.go

```diff
@@ -126,6 +130,37 @@ func NewCloud(cfg CloudConfig, metricsRegisterer prometheus.Registerer) (Cloud,
 	}, nil
 }
 
+func getVPC(metadata services.EC2Metadata, ec2Service services.EC2) string {
+
+	vpcId, err := metadata.VpcID()
+	if err == nil {
+		return vpcId
+	}
+
+	nodeName := os.Getenv("NODENAME")
+	if strings.HasPrefix(nodeName, "i-") {
```

Comment:
i haven't tried the new cloudProvider yet. What's the spec.providerID on the new cloudProvider? 
currently the spec.providerID is like ```aws:///us-west-2a/i-029d795cbe1d5c038```, if the new cloudprovider is having similar pattern, maybe we can support inferFromNode for other environment as well..
e.g. by 1. get the node object from node name, then get the providerID, then check the last component splitted by "/" is starting with `i-`(or use regex).

Reply:
Overall, I agree this is the better solution, but I consider this PR as a stepping stone towards that. It would require more rearranging of code to do k8s API calls and such as well. Especially if one wants to use the node watcher.

The providerID is the same as now when it comes to instance ID. But even now, it is not given that region is a part of the providerID and should rather be fetched from the topology labels.

Reply:
make sense, i'm fine to take this as a first step and we can have follow up PRs to further improve it.
Thanks for the info.